krabs/krabs/parser.hpp

```diff
@@ -367,14 +367,14 @@ namespace krabs {
         //
         //      The size of the TOKEN_USER structure differs
         //      depending on whether the events were generated on a 32 - bit
-        //      or 64 - bit architecture.Also the structure is aligned
+        //      or 64 - bit architecture. Also the structure is aligned
         //      on an 8 - byte boundary, so its size is 8 bytes on a
         //      32 - bit computer and 16 bytes on a 64 - bit computer.
         //      Doubling the pointer size handles both cases.
-        //
-        // So as we are 64-bit, this means we the SID starts at
-        // 16 bytes into the data
-        static const int sid_start = 16;
+        ULONG sid_start = 16;
+        if (EVENT_HEADER_FLAG_32_BIT_HEADER == (schema_.record_.EventHeader.Flags & EVENT_HEADER_FLAG_32_BIT_HEADER)) {
+            sid_start = 8;
```

Comment:
thank you for adding this! I wondered whether we needed special handling for 32-bit here.